pkg/providers/amifamily/al2.go

```diff
@@ -34,14 +34,29 @@ type AL2 struct {
 }
 
 // SSMAlias returns the AMI Alias to query SSM
-func (a AL2) SSMAlias(version string, instanceType *cloudprovider.InstanceType) string {
-	amiSuffix := ""
-	if !resources.IsZero(instanceType.Capacity[v1alpha1.ResourceNVIDIAGPU]) || !resources.IsZero(instanceType.Capacity[v1alpha1.ResourceAWSNeuron]) {
-		amiSuffix = "-gpu"
-	} else if instanceType.Requirements.Get(v1.LabelArchStable).Has(v1alpha5.ArchitectureArm64) {
-		amiSuffix = fmt.Sprintf("-%s", v1alpha5.ArchitectureArm64)
-	}
-	return fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2%s/recommended/image_id", version, amiSuffix)
+func (a AL2) SSMAlias(version string) map[string]scheduling.Requirements {
```

Comment:
Why not convert this function into something like `DefaultAMIs` and have it return the same thing the AMISelector is returning, a map of the AMI ids to requirements. 

That'll create consistency between the two different approaches.